frontend/server/src/DAO/Tags.php

```diff
@@ -77,15 +79,66 @@ public static function findPublicTagsByPrefix(
     }
 
     /**
-     * @return list<array{alias: string, total: int}>
+     * @return list<TagWithProblemCount>
      */
-    public static function getFrequentQualityTagsByLevel(
+    public static function getPublicQualityTagsByLevel(
         string $problemLevel
     ) {
         $sql = '
             SELECT
-                t.name AS alias,
-                COUNT(t.name) AS total
+                t.name AS name,
+                COUNT(t.name) AS problemCount
+            FROM
+                Problems_Tags pt
+            INNER JOIN
+                Tags t ON t.tag_id = pt.tag_id
+            INNER JOIN
+	            Problems p ON p.problem_id = pt.problem_id
+            WHERE
+                pt.problem_id IN (
+                    SELECT
+                        problem_id
+                    FROM
+                        Problems_Tags
+                    WHERE
+                        tag_id = (
+                            SELECT
+                                tag_id
+                            FROM
+                                Tags
+                            WHERE name = ?
+                        )
+                    ) AND
```

Comment:
nit:

```suggestion
                ) AND
```

en general se desea alinear los paréntesis al inicio de la línea en la cual se abre: en este caso debería ir alineado a L98.